packages/block-library/src/navigation-link/edit.js

```diff
@@ -232,16 +233,10 @@ export const updateNavigationLinkBlockAttributes = (
 		kind: newKind = originalKind,
 		type: newType = originalType,
 	} = updatedValue;
-	const normalizedTitle = title.replace( /http(s?):\/\//gi, '' );
+
 	const normalizedURL = url.replace( /http(s?):\/\//gi, '' );
-	const escapeTitle =
-		title !== '' &&
-		normalizedTitle !== normalizedURL &&
-		originalLabel !== title;
```

Comment:
so don’t escape if:

- title is empty
- the link already has a label which is not the title
- we obtained a title, so we don’t use the URL

probably the conditions here are to avoid double escaping? IDK.